fix: add rate limit and explain api key (#3)
jina-ai/meta-prompt: v4.txt

# Core principles

> 0. APIs require auth with bearer secret in the request header, i.e. "Authorization": "Bearer $JINA_API_KEY", this key can be found in https://jina.ai/?sui=apikey for free. Pull this JINA_API_KEY out and remind user to fill in.

Reviewer: ```suggestion
0. Assume, the Bearer is stored in the environment variable named "JINA_API_KEY" and add the following comment to the implementation: "Get your Jina AI API key for free: https://jina.ai/?sui=apikey".
```

Reviewer: So far, I had no cases where the bearer token was not sent with the request. Therefore I recommend removing this part of the instruction

For example, if you want to read the content of the page, you should extract the content from the response of the reader API like `content = reader_response["data"]["content"]`.
Another example, if you want to extract all the URL from a page, you can use the reader API with the "X-With-Links-Summary: true" header and then you can extract the links like `links = reader_response["data"]["links"]`.

4. Write the complete code, including input loading, calling the API functions, and saving/printing results;
Remember to use variables for required API keys, and point out to the user that they need to correctly set these variables.

5. Finally, Jina AI API endpoints rate limits:
Embedding & Reranker APIs (api.jina.ai/v1/embeddings, /rerank): 500 RPM & 1M TPM with API key; 2k RPM & 5M TPM with premium key
Reader APIs:

>  - r.jina.ai: 20 RPM without key, 200 RPM with key, 1k RPM premium

Reviewer: @hanxiao For simplification, should we enforce that the generated code always uses a key?
Same argument like for the zero-shot classifier. Simpler is better. The LLM already gets confused.
So I like the rate-limiting prompt, but I think we should remove the "without key" information

>  - r.jina.ai: 20 RPM without key, 200 RPM with key, 1k RPM premium

Reviewer: ```suggestion
 - r.jina.ai: 200 RPM, 1k RPM premium
```

>  - s.jina.ai: 5 RPM without key, 40 RPM with key, 100 RPM premium

Reviewer: ```suggestion
 - s.jina.ai: 40 RPM, 100 RPM premium
```

>  - g.jina.ai: No access without key, 10 RPM with key, 30 RPM premium

Reviewer: ```suggestion
 - g.jina.ai: 10 RPM, 30 RPM premium
```

Classifier APIs (api.jina.ai/v1/classify):

>  - Zero-shot: 200 RPM & 500k TPM with key; 1k RPM & 3M TPM premium

Reviewer: ```suggestion
 - 200 RPM & 500k TPM; 1k RPM & 3M TPM premium
```

>  - Few-shot: 20 RPM & 200k TPM with key; 60 RPM & 1M TPM premium

Reviewer: ```suggestion
```

> Classifier Training API (api.jina.ai/v1/train): 20 RPM & 200k TPM with key; 60 RPM & 1M TPM premium

Reviewer: ```suggestion
```

Reviewer: We only support zero shot at the moment for simplification.

> Segmenter API (segment.jina.ai): 20 RPM without key, 200 RPM with key, 1k RPM premium

Reviewer: ```suggestion
Segmenter API (segment.jina.ai): 200 RPM, 1k RPM premium
```
